src/tools/validations.ts

```diff
@@ -1,9 +1,50 @@
-import { GluegunToolbox } from "gluegun"
+import { GluegunToolbox, strings } from "gluegun"
 import { prefix } from "./pretty"
+const { pascalCase } = strings
+
+// #region Error Guards
+type IsError = (str: string) => boolean
+type ErrorMessage = (str?: string) => string
+type ErrorGuard = [IsError, ErrorMessage]
+
+const isIgnite: ErrorGuard = [
+  (str) => str.toLowerCase() === "ignite",
+  (str) => `Hey...that's my name! Please name your project something other than '${str}'.`,
+]
+const isOnlyNumbers: ErrorGuard = [
+  (str) => /^\d+$/.test(str),
+  () => `Please use at least one non-numeric character for your project name`,
+]
+const isKebabCase: ErrorGuard = [
+  (str) => str.includes("-"),
+  (str) =>
+    `Please use 'PascalCase', not 'kebab-case' for your project name. Ex: ${pascalCase(str)}`,
+]
+const isNotAlphaNumeric: ErrorGuard = [
+  (str) => !/^[a-z_][a-z0-9_]+$/i.test(str),
+  () =>
+    `The project name can only contain alphanumeric characters and underscore, but must not begin with a number.`,
+]
+
+const guards: ErrorGuard[] = [isIgnite, isOnlyNumbers, isKebabCase, isNotAlphaNumeric]
+
+/**
+ * check if the value matches any of the error guards
+ * @returns error message from the first guard that matches, or `true` if no guards match
+ */
+const validate = (value: string): true | string => {
+  for (const [isError, errorMessage] of guards) {
+    if (isError(value)) {
+      return errorMessage(value)
+    }
+  }
+  return true
+}
```

Comment:
[`GluegunPrompt` can accept a `validate` function](https://github.com/infinitered/gluegun/blob/master/src/toolbox/prompt-enquirer-types.ts#L14), that either returns `true` if it is a valid value, or a string if it is an error

Here is an demo of what that looks like:

https://user-images.githubusercontent.com/37849890/190262670-a849633a-59c7-4171-aaf3-29c4d9529aed.mov